### PR TITLE
update: support build for wasm

### PR DESCRIPTION
**Pull request by Rohith-Raju**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area build

/area CI

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR adds support to WASM target. This is a preliminary work for Google Summer of Code 
https://github.com/falcosecurity/evolution/issues/262

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: support build for wasm
```

### CODE REVIEW

**Comment by jasondellaluce** [comment]
This is ready for review! Me and @Rohith-Raju worked on this for the GSoC project. This is preliminary work for having a working Falco build for WASM.

cc @falcosecurity/libs-maintainers

**Comment by jasondellaluce** [comment]
Shoutout to @Rohith-Raju for the great work here! 🥳

**Comment by FedeDP** [comment]
Incredible job @Rohith-Raju and Jason! Thank you very much!
/milestone 0.12.0

File: userspace/libscap/emscripten/CMakeLists.txt
Comment (FedeDP) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: If affirmative, i'd wait until Grzeg's work is merged before going with this one, since he has been rebasing his PRs for months now :)

Comment (jasondellaluce) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: I need to check. Which of his PR concern is on top of your mind? These are simple stubs, so I'm happy to move them according to the new design or eventually help @gnosek integrate it in his patches in case we want to merge this first.

Comment (gnosek) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: 1/n (https://github.com/falcosecurity/libs/pull/1040) should be (mostly) fine since it's a no-op but merging only that is pointless
2/n (https://github.com/falcosecurity/libs/pull/1041) will almost certainly conflict
3/n (https://github.com/falcosecurity/libs/pull/1042) might be fine but since it builds on top of 2/n, it won't apply cleanly either

BTW, from a drive-by look, it seems `emscripten` will fit right in in the platform work (either as a separate platform, or possibly as a generic platform instance that does not list processes, users etc.)

Comment (jasondellaluce) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: @gnosek do you prefer to merge yours first and then let us adapt this one accordingly, or the contrary?

Comment (gnosek) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: Does the order imply *who* will do the adapting? ;D

Comment (jasondellaluce) [reply userspace/libscap/emscripten/CMakeLists.txt:3]: @gnosek I'm ok in merging yours first then :)

**Comment by FedeDP** [comment]
Moving to libs-backlog since this won't be part of 0.12.0.
/milestone libs-backlog

File: userspace/libscap/scap.c
Comment (FedeDP) [inline userspace/libscap/scap.c:189]: Why is this needed? I mean, i think part of the @gnosek purpose when doing the platform refactor was to avoid conditionals like this one.
Also, this seems more like a fix unrelated to the PR, right?

File: cmake/modules/libsinsp.cmake
Comment (FedeDP) [inline cmake/modules/libsinsp.cmake:33]: This is a rather interesting change; can you explain why was it needed?

**Review by FedeDP — COMMENTED** [review]
Great job! Left a couple of comments, but PR LGTM!

File: userspace/libscap/scap.c
Comment (jasondellaluce) [reply userspace/libscap/scap.c:189]: Seems like Grzegz implemented the same logic in https://github.com/falcosecurity/libs/blob/d1f3bd3a5d51b89e90030ff666967f866a1e49f9/userspace/libscap/scap.c#L38C50-L38C50. This is a leftover, good catch! Let me remove it.

**Comment by jasondellaluce** [comment]
The `Driver SCHEMA_VERSION checks` failure is a false positive here.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]

**Pull request merged**
